### PR TITLE
docs(DiseasyActivity): improve documentation

### PR DESCRIPTION
**Pull request by RasmusSkytte**
### Intent
This PR is part of the re-factoring of PR #30 which got too large due to scope-creep.
I am breaking that PR up into smaller chunks to make reviewing easier.

This PR does the following
* Documentation was added according to documentation standard
* TODOs converted to issues on github

### Approach
N/A

### Known issues
* [x] The documentation contains a lot of "TODO"s which should be resolved before merging.

### Checklist

* [x] The PR passes all local unit tests
* [ ] I have documented any new features introduced
* [ ] If the PR adds a new feature, please add an entry in `NEWS.md`
* [ ] A reviewer is assigned to this PR

### CODE REVIEW

File: R/DiseasyActivity.R [hunk 1]
```diff
@@ -39,12 +81,24 @@ DiseasyActivity <- R6::R6Class( # nolint: object_name_linter
...
-        work_risk <- aggregate(faWork ~ date, data = mg_reference_scenario, FUN = mean)
-        self$change_risk(date = work_risk$date, type = "work", risk = work_risk$faWork)
+
+        # Use the Danish activity units
+        self$set_activity_units(dk_activity_units)
+
+        # Use the Danish restrictions
+        self$change_activity(dk_reference_scenario)
+
+        # TODO: Lasse, is this correct?
```
Comment (RasmusSkytte) [inline R/DiseasyActivity.R:93]: TODO to be resolved

File: R/DiseasyActivity.R [hunk 2]
```diff
@@ -39,12 +81,24 @@ DiseasyActivity <- R6::R6Class( # nolint: object_name_linter
...
+        # TODO: wouldn't it be nice if this happened inside $change_activity?
+        # TODO: this way, the `dk_reference_scenario` could be loaded in a single function?
```
Comment (RasmusSkytte) [inline R/DiseasyActivity.R:95]: TODO to be resolved

File: R/DiseasyActivity.R [hunk 3]
```diff
@@ -39,12 +81,24 @@ DiseasyActivity <- R6::R6Class( # nolint: object_name_linter
...
+        # The "social_distance_work" parameter varies across activity units. If several activity units are active
+        # on the same date, we compute the mean "social_distance_work" use this risk for all units on that date
+        # TODO: should this mean be weighted the "size" of the activity units?
```
Comment (RasmusSkytte) [inline R/DiseasyActivity.R:98]: TODO to be resolved

File: R/DiseasyActivity.R [hunk 4]
```diff
@@ -378,27 +439,26 @@ DiseasyActivity <- R6::R6Class( # nolint: object_name_linter
...
 
-      openness <- self$get_scenario_openness()
+      contacts <- openness <- self$get_scenario_openness()
 
-      contacts <- openness
       # Apply .risk_matrix
-      for (dd in seq_along(openness)) { #looping over dates
+      for (dd in seq_along(openness)) { # looping over dates
         for (tt in private$activity_types) {
           # TODO: genWeight
```
Comment (RasmusSkytte) [inline R/DiseasyActivity.R:447]: TODO to be resolved

File: R/0_documentation.R [hunk 5]
```diff
@@ -15,6 +23,22 @@ rd_case_definition <- function(type = "param") {
 }
 
 
+rd_contact_basis <- function(type = "param") {
+  checkmate::assert_choice(type, c("param", "field"))
+  paste("(`list(list())`)\\cr",
+        "A nested list with all the needed information for the contact_basis\\cr",
+        "* `counts` contains the age stratified contact counts across the arenas of the basis",
+        "  (e.g. 'work', 'home', 'school', 'other')\\cr",
+        "* `prop` contains a list of the proportion of population in 5-year age-groups\\cr",
```
Comment (LasseEngboChr) [inline R/0_documentation.R:32]: I do not find that we should specify 5yr groups - it is 'just' the default in what we supply with the package. Suggest: "... population in each age group."

File: R/DiseasyActivity.R [hunk 6]
```diff
@@ -1,8 +1,52 @@
...
+#'   The `DiseasyActivity` module is responsible for handling the societal activity.
+#'   By providing a `contact_basis`, `activity_units` and a `scenario` to the module, the
+#'   module provides activity matrices throughout the scenario (with flexible age-groupings)
+#'   as well as the overall "activity" level in society.
+#'
+#'   The `contact_basis` contains information about:
+#'     - contact matrices (contact rates between age groups)
+#'     - population (size and proportion of population in age groups)
+#'
+#'   The `activity_units` projects restrictions into smaller "units" that are independently
```
Comment (LasseEngboChr) [inline R/DiseasyActivity.R:13]: ... restrictions, guidelines and other changes in activity intro smaller ...

File: R/DiseasyActivity.R (hunk 1 above)
Comment (LasseEngboChr) [reply R/DiseasyActivity.R:93]: Indeed

File: R/DiseasyActivity.R (hunk 2 above)
Comment (LasseEngboChr) [reply R/DiseasyActivity.R:95]: I find it usefull to split the setting of activity units - all that are allowed - and the actual opening and closing of those. If there were more reference scenarios with individual activity units then one could consider making an internal function that sets both the units and the reference changes in activity. That could be next version ...

File: R/DiseasyActivity.R (hunk 3 above)
Comment (LasseEngboChr) [reply R/DiseasyActivity.R:98]: I suggest that we leave this for a future discussion. Then we can make it a feature update.

File: R/DiseasyActivity.R [hunk 7]
```diff
@@ -254,15 +304,14 @@ DiseasyActivity <- R6::R6Class( # nolint: object_name_linter
         coll$push("scenario_matrix not set. Invoke change_activity() first.")
       }
       checkmate::assert_character(type, pattern = "(work)|(school)|(home)|(other)", any.missing = FALSE, add = coll)
-      checkmate::assert_numeric(risk, lower = 0, add = coll)
+      checkmate::assert_numeric(risk, lower = 0, add = coll) # TODO: upper is 1, right?
```
Comment (LasseEngboChr) [inline R/DiseasyActivity.R:307]: No, it just have to be non negative (Using risk = 0 is technically OK but probably not so relevant ...).

File: R/DiseasyActivity.R [hunk 8]
```diff
@@ -347,7 +398,8 @@ DiseasyActivity <- R6::R6Class( # nolint: object_name_linter
       openness <- lapply(list_active, private$add_activities)
 
       # Apply .risk_matrix
-      for (dd in seq_along(openness)) { #looping over dates
+      # TODO: is this not already applied through "add_activities"?
```
Comment (LasseEngboChr) [inline R/DiseasyActivity.R:401]: NTS: Get back to this

File: R/DiseasyActivity.R [hunk 9]
```diff
@@ -510,45 +575,75 @@ DiseasyActivity <- R6::R6Class( # nolint: object_name_linter
...
-    # Each element is a list of open and a list of closed activity units from the corresponding date
-    scenario = list(),
 
-    activity_units = list(),
+    # @field direction (`character(1)`)\cr
+    # The "direction" of the changes to activity
+    # The direction can be either "opening" or "closing"
+    # If "opening", the activity units are added to a fully closed starting point
+    # If "closing", the activity units are subtracted from a fully open starting point
+    # TODO: Lasse, can you confirm?
```
Comment (LasseEngboChr) [inline R/DiseasyActivity.R:597]: I can

File: R/DiseasyActivity.R [hunk 10]
```diff
@@ -510,45 +575,75 @@ DiseasyActivity <- R6::R6Class( # nolint: object_name_linter
...
+    direction = NULL,
+
+    # @field upper_activity_level (`numeric(1)`)\cr
+    # The upper level of activity. Used as a safe guard to ensure the combination of
+    # activity units does not exceed (drop below) the maximum (minimum) activity associated
+    # with the fully open (closed) starting point.
+    # Takes a value 1 when direction is "opening" and 0 when direction is "closing"
+    # TODO: Lasse, can you confirm?
```
Comment (LasseEngboChr) [inline R/DiseasyActivity.R:605]: Indeed

File: R/DiseasyActivity.R [hunk 11]
```diff
@@ -510,45 +575,75 @@ DiseasyActivity <- R6::R6Class( # nolint: object_name_linter
...
 
-    # Parameters for contact_basis
-    .contact_basis = list(counts = NULL, # list of matrices of number of contacts per individual per day ('m')
-                          prop = NULL, # Vector of population per age group in 'counts'
-                          pop = NULL,
-                          description = NULL),
+    # @field .contact_basis (`data.frame`)\cr
+    # The contact_basis used to project the activity units in
+    # Must have columns "counts", "prop", "pop_ref_1yr", "description"
+    .contact_basis = NULL,
```
Comment (LasseEngboChr) [inline R/DiseasyActivity.R:633]: Check

File: R/DiseasyActivity.R [hunk 12]
```diff
@@ -589,12 +687,15 @@ DiseasyActivity <- R6::R6Class( # nolint: object_name_linter
...
+    },
+
 
     # Generate symmetric weight matrix based on diagonal.
-    # Ensures that you can add so that when the diagonal is one then the rest is as well
+    # @description
+    #   Ensures that you can add so that when the diagonal is one then the rest is as well
     #   it is done by adding contacts to and from a given age group to those that are younger
-    # (In Danish this could be labelled as "sildebensparket")
+    #   (In Danish this could be labelled as "sildebensparket")
```
Comment (LasseEngboChr) [inline R/DiseasyActivity.R:697]: Suggestion: "(Like herringbone flooring)"

Comment (LasseEngboChr) [reply R/DiseasyActivity.R:697]: NB: this function was previously named genWeight ...

File: R/DiseasyActivity.R [hunk 13]
```diff
@@ -398,19 +450,25 @@ DiseasyActivity <- R6::R6Class( # nolint: object_name_linter
...
-          # TODO: genWeight
+          # The openness (i.e. the degree to which each age-group has their contacts restricted) are converted to a
+          # "herringbone" pattern and multiplied elementwise to the baseline contact matrices.
+          # By converting to the "herringbone" pattern, the age-stratified activity reductions are applied using an
+          # assumption that reductions are absolute. e.g. if a person has their activity restricted by 50% then their
+          # contacts are also restricted by 50 %.
+          # In contrast, one could assume that reductions are multiplicative in nature. e.g. if age-group i is
+          # restricted by 50 % and age-group j is restricted by 10 %, then contacts between age-groups i and j would be
+          # reduced by 1 - 0.5 * 0.9 = 55 %.
           contacts[[dd]][[tt]] <- private$vector_to_matrix(openness[[dd]][[tt]]) * self$contact_basis$counts[[tt]]
```
Comment (RasmusSkytte) [inline R/DiseasyActivity.R:466]: @LasseEngboChr  Last thing I need you to take a look at.

**Review by LasseEngboChr — APPROVED** [review]

**Pull request merged**